Notification Service/NotificationService.swift

```diff
@@ -74,70 +88,74 @@ class NotificationService: UNNotificationServiceExtension {
         }
     }
 
-    private func getConfiguredChatClient() -> ChatClient? {
-        guard let user = User.current() else { return nil }
+    private func getConfiguredChatClient() async -> ChatClient? {
+        guard let user = User.current(), user.isAuthenticated  else { return nil }
 
-        var config = ChatClientConfig(apiKey: .init("hvmd2mhxcres"))
+        var config = ChatClientConfig(apiKey: .init(Config.shared.environment.chatAPIKey))
         config.isLocalStorageEnabled = true
         config.applicationGroupIdentifier = Config.shared.environment.groupId
+        let client = ChatClient(config: config, tokenProvider: nil)
+
+        do {
+            // Get the app token and then apply it to the chat client.
+            let result: String = try await withCheckedThrowingContinuation { continuation in
+                PFCloud.callFunction(inBackground: "getChatToken",
+                                     withParameters: [:]) { (object, error) in
+
+                    if let error = error {
+                        continuation.resume(throwing: error)
+                    } else if let value = object as? String {
+                        continuation.resume(returning: value)
+                    } else {
+                        continuation.resume(throwing: ClientError.apiError(detail: "Request failed"))
+                    }
+                }
+            }
 
-        let client = ChatClient(config: config)
-        let token = Token.development(userId: user.objectId!)
-        client.setToken(token: token)
+            let token = try Token(rawValue: result)
+            client.setToken(token: token)
```

Comment:
Our cloud convenience functions aren't in this extension target so I'm calling GetToken manually.